pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -968,3 +974,45 @@ func (h *upgradeHandler) startVM(ctx context.Context, vm *kubevirtv1.VirtualMach
 		Do(ctx)
 	return res.Error()
 }
+
+// checkLogReadyCondition times out LogReady condition, thus moving on to the next step if upgradeLogCondLogReadyTimeout
+// time has passed since UpgradeLog's logging infrastructure setup started
+func (h *upgradeHandler) checkLogReadyCondition(upgrade *harvesterv1.Upgrade) (*harvesterv1.Upgrade, error) {
+	upgradeConfig, err := settings.DecodeConfig[settings.UpgradeConfig](settings.UpgradeConfigSet.Get())
+	if err != nil {
+		return upgrade, err
+	}
+	timeout := upgradeConfig.LogReadyTimeout
+
+	ts := harvesterv1.LogReady.GetLastUpdated(upgrade)
+	t, err := time.Parse(time.RFC3339, ts)
+	if err != nil {
+		logrus.Errorf("Failed to parse last updated time: %v", err)
+		return upgrade, nil
+	}
+	if time.Since(t) >= timeout {
+		toUpdate := upgrade.DeepCopy()
+		upgradeLog, err := h.upgradeLogCache.Get(util.HarvesterSystemNamespaceName, upgrade.Status.UpgradeLog)
+		if err != nil {
+			logrus.Errorf("Failed to get upgrade log: %v", err)
+			return upgrade, nil
+		}
+		logrus.Infof("Timed out waiting for UpgradeLog %q's %q condition to pass", upgradeLog.Name, harvesterv1.LogReady)
+
+		toUpdateUpgradeLog := upgradeLog.DeepCopy()
+		harvesterv1.InfraReady.SetStatus(toUpdateUpgradeLog, string(metav1.ConditionFalse))
+		harvesterv1.InfraReady.Reason(toUpdateUpgradeLog, "Timeout")
+		harvesterv1.InfraReady.Message(toUpdateUpgradeLog, "Timed out creating logging infrastructure")
+		_, err = h.upgradeLogClient.Update(toUpdateUpgradeLog)
+		if err != nil {
+			logrus.Errorf("Failed to update upgradeLog: %v", err)
+			return upgrade, nil
+		}
+		setLogReadyCondition(toUpdate, corev1.ConditionFalse, "Timeout", "Timed out creating logging infrastructure")
+		return h.upgradeClient.Update(toUpdate)
+	}
+
+	logrus.Info("Waiting for LogReady condition to be set")
```

Comment:
change to `logrus.Debug` to reduce the log

Reply:
Done.